src/jit/morph.cpp

```diff
@@ -12739,20 +12713,13 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
         {
             /* "x ^ -1" is "~x" */
             
-            if ((op2->gtOper == GT_CNS_INT) && (op2->gtIntConCommon.IconValue() == -1))
+            if (op2->IsIntegralConst(-1))
             {
                 tree->ChangeOper(GT_NOT);
                 tree->gtOp2 = NULL;
                 DEBUG_DESTROY_NODE(op2);
             }
-            else if ((op2->gtOper == GT_CNS_LNG) && (op2->gtIntConCommon.LngValue() == -1))
-            {
-                tree->ChangeOper(GT_NOT);
-                tree->gtOp2 = NULL;
-                DEBUG_DESTROY_NODE(op2);
-            }
-            else if ((op2->gtOper == GT_CNS_INT) && (op2->gtIntConCommon.IconValue() == 1) &&
-                     op1->OperIsCompare())
+            else if (op2->IsIntegralConst(1) && op1->OperIsCompare())
```

Comment:
Yes, this one I had noted, but also did not result in any diffs. I'm not certain why the xor of -1 is handled for both int and long, but not xor of 1.